docs/blueprints/EP038.rst

```diff
@@ -0,0 +1,460 @@
+
+*****************************************
+EP03X - Broadcasting events
+*****************************************
+
+########
+Abstract
+########
+
+This blueprint details how Kytos should export/broadcast its events to external applications safely and scalable. 
+
+##########
+Motivation
+##########
+
+With Kytos running in production, we learned that Kytos events can be leveraged for multiple purposes, from capacity planning to topology evaluation, from troubleshooting to accounting. However, creating Kytos napps to only monitor the environment should be avoided to minimize possible impacts on running the kytos instance. 
+Moreover, this NApp can expedite the prototyping of new applications and frameworks that need an understanding the network topology and control plane, such as the AtlanticWave-SDX project. 
+Based on the information of the AtlanticWave-SDX project, the system design integrates the Kytos-ng SDN platform and the SDX Local Controller SDX-LC. 
+This integration aims to fulfill the functionalities required by the AtlanticWave-SDX project. 
+
+Here is an overview of the AtlanticWave-SDX project components:
+
+:Kytos-ng Topology: 
+Integrating with the Kytos-ng framework provides a convenient API for retrieving the OXPO topology. This design choice leverages Kytos-ng's capabilities and simplifies obtaining topology data from the SDN platform. To fetch the topology data, the SDX-LC CRON function can request GET or pull through the endpoint: 
+
+"http://localhost:8181/api/kytos/topology/v3/".
+
+:Kytos Event Handling: 
+Kytos-ng uses Kytos Event decorated functions within the Kytos-sdx-topology Napp to handle specific events related to topology changes. 
+This event-driven approach enables the system to respond promptly and capture changes in the Kytos topology. It also allows for flexibility in managing different types of events and provides a scalable solution.
+
+:Kytos-sdx-topology Napp: 
+The Kytos-sdx-topology Napp should actively monitor topology changes by subscribing to events using the Kytos Event decorated function. This design choice must ensure the system stays updated with any modifications in the Kytos-ng topology and enables real-time processing of topology-related events.
+
+Kytos-sdx-topology kytos Napp
+https://github.com/atlanticwave-sdx/kytos-sdx-topology
+
+Actual kytos-sdx-topology napp is designed to do the whole process in one napp, In atlanticwave sdx infrastructure, it's crucial to monitor the status of interdomain links for connectivity and reliability. This use case involves setting up a system to detect link status changes (up/down) and notify consumers about these events. 
+
+The purpose of this blueprint is to design and evaluate the introduction of a message queue and separate the process as follows:
+
+Creating two NApps to separate the event listener and treatment from the post office message broker simplifies the system architecture, enhances modularity, and improves fault tolerance. Here's how:
```

Comment:
Luis, this part is a bit confusing. Maybe you're already picturing that you'll have a consumer NApp on a different kytosd instance. The consumer on the other end could be any kind of consumer whether a NApp on another instance, or BAPM, or whatever else application, or even another application in a different language (and that's why I'm also asking to specify the serialization that will be used, whether or not the consumer will be needed to be in Python or not):

![20240405_105626](https://github.com/kytos-ng/kytos/assets/1010796/0a293622-b28d-46cf-b1bc-74624f89da51)

For whoever is reading this doc, the "two NApps" here are kind of not making it clear, and as if this NApp here were to pretty much act as a parallel event bus, which it can be, but the thing is, if one day we decide to also support a richer message broker to send the events it'll be baked in the core and not in a NApp. So, for this NApp here, as you're documenting, it's a adapt to bridge events from the internal bus to an external (RabbitMQ) message broker. 

That said, this spec should focus on the producer, of course you can also briefly hightlight how a consumer would look like it you want to use a consumer in the SDX, but users shouldn't be constrained to always use "two NApps" to have events being bridged to RabbitMQ, just a single NApp that you're specifying should suffice. Now, if you with to make official the consumer in another blueprint related to SDX, it's up to you. That's probably already clear to you, but just highlighting this point here.